### PR TITLE
add guest as url param for shared events, switch schema generation to drf-spectacular

### PR DESCRIPTION
**Pull request by jeriox**

### CODE REVIEW

File: ephios/api/urls.py [hunk 1]
```diff
@@ -91,13 +90,13 @@
...
-        get_schema_view(
-            title="ephios",
-            description="ephios API",
-            version=settings.EPHIOS_VERSION,
-            url=settings.GET_SITE_URL(),
-        ),
+        SpectacularAPIView.as_view(),
         name="openapi-schema",
     ),
+    path(
```
Comment (jeriox) [inline ephios/api/urls.py:96]: @felixrindt the built-in schema generation of DRF is deprecated (https://www.django-rest-framework.org/api-guide/schemas/), they recommend drf-spectacular. I rebuilt everything as is, we can decide whether we want to have a swagger view here

Comment (felixrindt) [reply ephios/api/urls.py:96]: I really like the swagger view! Can it show the correct base url for the example pagination links?

File: ephios/plugins/federation/models.py [hunk 2]
```diff
@@ -183,10 +183,16 @@ def all_participations(self):
...
+        return reverse(
+            "federation:shift_signup",
+            kwargs={"pk": shift.pk, "guest": self.federated_user.federated_instance.pk},
+        )
 
     def reverse_event_detail(self, event):
-        return reverse("federation:event_detail", kwargs={"pk": event.pk})
+        return reverse(
+            "federation:event_detail",
+            kwargs={"pk": event.pk, "guest": self.federated_user.federated_instance.pk},
```
Comment (felixrindt) [inline ephios/plugins/federation/models.py:194]: No problem here, but did you know that using
```suggestion
            kwargs={"pk": event.pk, "guest": self.federated_user.federated_instance_id},
```
causes one db query less?

**Review by felixrindt — APPROVED** [review]

File: ephios/api/urls.py (hunk 1 above)
Comment (jeriox) [reply ephios/api/urls.py:96]: I couldn't find a way to customize it, I think it is intentional as this is only fixed text.
When you use the "try it out" feature, it displays the correct pagination links

**Pull request merged**